### PR TITLE
add preview oval

### PR DESCRIPTION
**Pull request by songfei**
implemented this issue #18

### CODE REVIEW

File: lib/src/ui.dart [hunk 1]
```diff
@@ -1,10 +1,15 @@
-import 'dart:ui' as ui;
 import 'dart:math';
+import 'dart:ui' as ui;
 
 import 'package:crop/src/geometry_helper.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/rendering.dart';
 
+enum Clipper {
```
Comment (xclud) [inline lib/src/ui.dart:8]: I suggest `CropShape` as the enum name.

File: lib/src/ui.dart [hunk 2]
```diff
@@ -16,6 +21,7 @@ class Crop extends StatefulWidget {
   final Widget helper;
   final Widget overlay;
   final bool interactive;
+  final Clipper clipper;
```
Comment (xclud) [inline lib/src/ui.dart:24]: I suggest `shape` as the name.

**Comment by songfei** [comment]
modify the variable name

**Comment by xclud** [comment]
@songfei i run your PR on my local device. It turned out the changes you've made are not enough to make the oval shape work. Specially in `CropRenderObjectWidget` and its `updateRenderObject` some changes are needed.

I completed the missing part and the oval shape is working now. I close this PR without merging t, but i credit you in the CHANGELOG.md as i am inspired by your PR.

Thank You.

**Comment by songfei** [comment]
@xclud  I have used your components in my app. Thank you very much for your work.

**Pull request closed without merging**